danny-avila/LibreChat
# 🛠️ fix: Tool Filtering in PluginsClient

I implemented tool filtering in the PluginsClient and added comprehensive unit tests to ensure its functionality. I also performed some code cleanup by removing unused components.

This closes #3259

- Added functionality to filter tools in PluginsClient based on `filteredTools` and `includedTools` arrays in the request's app locals
- Implemented logic to prioritize `includedTools` over `filteredTools` for enhanced flexibility in tool selection
- Created a comprehensive test suite covering various scenarios of tool filtering, including:
  - Filtering out specific tools
  - Including specific tools
  - Prioritization of `includedTools` over `filteredTools`
  - Behavior when no filters are provided
- Removed the unused legacy Conversation component to clean up the codebase
- Updated relevant imports to maintain code consistency

## Testing

I thoroughly tested the new tool filtering feature using the newly created unit tests. These tests cover a wide range of scenarios to ensure the feature works as expected under different conditions.

To test this feature:

1. Run the unit tests for the PluginsClient
2. Manually test the feature by making requests with different combinations of `filteredTools` and `includedTools` in the app locals
3. Verify that the correct tools are included or excluded based on the provided filters

### Test Configuration:

- Node.js environment
- Jest testing framework

## Checklist
